skopt/sample.py

```diff
@@ -0,0 +1,75 @@
+from collections import Mapping
+from numbers import Integral
+from numbers import Real
+
+from scipy.stats._distn_infrastructure import rv_frozen
+from scipy.stats.distributions import randint
+from scipy.stats.distributions import uniform
+
+from sklearn.utils import check_random_state
+from sklearn.utils.fixes import sp_version
+
+
+class Categorical:
+    def __init__(self, *args):
+        self.values = args
+        self.low = min(self.values)
+        self.high = max(self.values)
+        self._rvs = randint(0, len(self.values))
+
+    def rvs(self, random_state=None):
```

Comment:
Can you move the `random_state` version check from below to here? So that doing `Categorical([1, 2, 3, 4]).rvs()` does not raise an error in older versions of SciPy?